projects/suricata/build.sh

```diff
@@ -0,0 +1,271 @@
+export CARGO_BUILD_TARGET="x86_64-unknown-linux-gnu"
+
+(
+cd pcre2-10.39
+./configure --disable-shared
+make -j$(nproc) clean
+make -j$(nproc) all
+make -j$(nproc) install
+)
+
+(
+cd lz4-1.9.2
+make liblz4.a
+cp lib/liblz4.a /usr/local/lib/
+cp lib/lz4*.h /usr/local/include/
+)
+
+(
+cd jansson-2.12
+./configure --disable-shared
+make -j$(nproc)
+make install
+)
+
+(
+cd fuzzpcap
+mkdir build
+cd build
+cmake ..
+make install
+)
+
+(
+cp -r libhtp suricata && cd suricata
+
+sh autogen.sh
+./src/tests/fuzz/oss-fuzz-configure.sh && make -j$(nproc)
+
+mkdir libfuzzer
+
+./src/suricata --list-app-layer-protos | tail -n +2 | while read i; do cp src/fuzz_applayerparserparse libfuzzer/fuzz_applayerparserparse_$i; done
+
+mv libfuzzer/fuzz_applayerparserparse_http libfuzzer/fuzz_applayerparserparse_http1
+
+cp src/fuzz_applayerprotodetectgetproto libfuzzer/fuzz_applayerprotodetectgetproto
+cp src/fuzz_mimedecparseline libfuzzer/fuzz_mimedecparseline
+cp src/fuzz_predefpcap_aware libfuzzer/fuzz_predefpcap_aware
+cp src/fuzz_sigpcap_aware libfuzzer/fuzz_sigpcap_aware
+
+make clean
+)
+
+echo "[x] AFL++ stage"
+export CC=afl-clang-lto
+export CXX=afl-clang-lto++
+export ac_cv_func_malloc_0_nonnull=yes
+export ac_cv_func_realloc_0_nonnull=yes
+export CFLAGS="-g -fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+export CXXFLAGS="-g -fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+export RUSTFLAGS="-C debug-assertions -C overflow_checks -C passes=sancov-module -C codegen-units=1 -C llvm-args=-sanitizer-coverage-level=3 -C llvm-args=-sanitizer-coverage-trace-pc-guard -C llvm-args=-sanitizer-coverage-prune-blocks=0 -C opt-level=3 -C target-cpu=native -Clink-arg=-fuse-ld=gold"
+
+(
+cd pcre2-10.39
+./configure --disable-shared
+make -j$(nproc) clean
+make -j$(nproc) all
+make -j$(nproc) install
+)
+
+(
+cd lz4-1.9.2
+make liblz4.a
+cp lib/liblz4.a /usr/local/lib/
+cp lib/lz4*.h /usr/local/include/
+)
+
+(
+cd jansson-2.12
+make clean
+./configure --disable-shared
+make -j$(nproc)
+make install
+)
+
+(
+cd fuzzpcap
+rm -rf build
+mkdir build
+cd build
+cmake ..
+make install
+)
+
+(
+cp -r libhtp suricata && cd suricata
+sh autogen.sh
+./src/tests/fuzz/oss-fuzz-configure.sh && make -j$(nproc)
+
+mkdir afl
+
+./src/suricata --list-app-layer-protos | tail -n +2 | while read i; do cp src/fuzz_applayerparserparse afl/fuzz_applayerparserparse_$i; done
+
+mv afl/fuzz_applayerparserparse_http afl/fuzz_applayerparserparse_http1
+
+cp src/fuzz_applayerprotodetectgetproto afl/fuzz_applayerprotodetectgetproto
+cp src/fuzz_mimedecparseline afl/fuzz_mimedecparseline
+cp src/fuzz_predefpcap_aware afl/fuzz_predefpcap_aware
+cp src/fuzz_sigpcap_aware afl/fuzz_sigpcap_aware
+make clean
+)
+
+unset CFLAGS
+unset CXXFLAGS
+unset LDFLAGS
+unset RUSTFLAGS
+
+echo "[x] Sydr stage"
+export CC=clang
```

Comment:
CXX, CFLAGS, CXXFLAGS?

Reply:
There were CXX, LDFLAGS, and RUSTFLAGS set during previous build steps. Could they affect sydr-build?

Reply:
yep, my fault